ibis/backends/tests/test_join.py

```diff
@@ -205,6 +206,11 @@ def test_semi_join_topk(batting, awards_players):
     raises=exc.IbisTypeError,
     reason="DuckDB as of 0.7.1 occasionally segfaults when there are `null`-typed columns present",
 )
+@pytest.mark.broken(
+    ["snowflake"],
+    reason="Case isn't preserved for memtable yet",
+    raises=sa.exc.ProgrammingError,
```

Comment:
Do you have access to the SQL query that is causing this problem?
From what I noticed, Snowflake has a bug in handling the VALUES syntax. Quoting is not correctly from column names.

The SQL query below works in most SQL engines
```
SELECT "MEMTABLE"."col1"
FROM (VALUES ('a')) AS "MEMTABLE" ("col1");
```
but Snowflake expects the following query:
```
SELECT "MEMTABLE"."""col1"""
FROM (VALUES ('a')) AS "MEMTABLE" ("col1")
```
As a workaround, we can use the following syntax.
```
SELECT $1 as "col1"
FROM (VALUES (:param_1))
```
To achieve this, we need to omit the name parameter from the `sa.values` call.


https://github.com/ibis-project/ibis/blob/8dcfb8d7cb7b84ffe1f8cd42a09be27b77d7d777/ibis/backends/base/sql/alchemy/query_builder.py#LL124

Reply:
Can you open a follow-up PR ti fix this?